src/abstract_value.rs

```diff
@@ -102,6 +173,13 @@ impl AbstractValue {
             value: self.value.widen(&other.value, &join_condition.value),
         }
     }
+
+    /// Returns a clone of the value with the given span prepended to its provence.
+    pub fn with_provenance(&self, provenance: Span) -> AbstractValue {
+        let mut result = self.clone();
```

Comment:
I think as a general rule you want to take self and let the caller decide if they need to clone or not. But in this case the ergonomics of that might be uglier.

Reply:
I tried to do just that, but the borrow checker was unhappy about it.